### ISSUE TITLE
Deleting DigitalOcean droplets fails for terraform destroy

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-digitalocean#14]
_This issue was originally opened by @aknuds1 as hashicorp/terraform#14329. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version
0.9.4

### Affected Resource(s)
- digitalocean_droplet

### Terraform Configuration Files
[Tectonic Installer Terraform config for DigitalOcean](https://github.com/aknuds1/tectonic-installer/tree/e4cd2ae4028c3b409d39ff45faa8346d5e084b83/platforms/digitalocean)

### Debug Output
[Debug log gist](https://gist.github.com/aknuds1/7ea46bd557f5ed11c36cedd8760f74f5)

### Expected Behavior
I am trying to delete my DigitalOcean droplets created with Terraform, via `terraform destroy`.

### Actual Behavior
Terraform bails with an error per droplet it's trying to delete: `digitalocean_droplet.etcd_node.0: Error deleting droplet: invalid character 'S' looking for beginning of value`.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Clone the [Tectonic Installer repository](https://github.com/aknuds1/tectonic-installer/commit/e4cd2ae4028c3b409d39ff45faa8346d5e084b83) into `$GOPATH/src/github.com/coreos/tectonic-installer`
2. Install Yarn
3. Create DigitalOcean droplets: `PLATFORM=digitalocean make apply`
4. Delete DigitalOcean droplets: `PLATFORM=digitalocean make destroy`

**Comment by keinohguchi** [comment]
Thank you for your report, @aknuds1!

Yeah, I saw this error last night with different resource, digitalocean_firewall.

Let me double check and get back here.

Cheers!

**Comment by keinohguchi** [comment]
Here is what I got today, with the latest DO provider:

```
terraform destroy -force
data.terraform_remote_state.main: Refreshing state...
digitalocean_firewall.client_firewall: Refreshing state... (ID: a9e53891-c8f0-4f5a-a0b5-e
d869695f85d)
digitalocean_firewall.server_firewall: Refreshing state... (ID: 83d0de1e-4424-486f-9d20-d
098f3a828b3)
digitalocean_firewall.client_firewall: Destroying... (ID: a9e53891-c8f0-4f5a-a0b5-ed86969
5f85d)
digitalocean_firewall.server_firewall: Destroying... (ID: 83d0de1e-4424-486f-9d20-d098f3a
828b3)
digitalocean_firewall.server_firewall: Destruction complete
Error applying plan:

1 error(s) occurred:

* digitalocean_firewall.client_firewall (destroy): 1 error(s) occurred:

* digitalocean_firewall.client_firewall: Error deleting firewall: invalid character 'S' l
ooking for beginning of value

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

```
$ terraform -v
Terraform v0.10.0-dev (e50a57bcb7f3e7a0469f5b9b030df9074bb10bfa+CHANGES)
```
Here is the actual `main.tf` which caused this error, but I don't think it's specific to this, as when I rerun this, it went successfully.

https://github.com/keinohguchi/do-in-action/blob/master/firewalls/both/000-deny-all/main.tf

I'll keep an eye on it.

**Comment by andrewsomething** [comment]
```
digitalocean_droplet.etcd_node.0: Error deleting droplet: invalid character 'S' looking for beginning of value.
```

This was caused by issues further down the stack. That error comes from godo which Terraform passes straight through. The `S` is likely `Server Error` which was incorrectly returned by the DigitalOcean API.

I'm going to go ahead and close this one out as godo now handles this much better, and we've just pulled in an update to the latest version. See:https://github.com/digitalocean/godo/pull/156

Thanks for baring with us!
